5.2/Piece of cake.py

```diff
@@ -0,0 +1,36 @@
+"""
+Calculate price for recipe's ingredients
+"""
+
+
+def get_recipe_price(prices: dict, optionals: list = None, **gram_ingredients: dict) -> float or None:
+    """Receives dictionary when key its ingredients name's and item is their price for 100 g,
+    list of optional ingredients that someone don't want to buy at all and dictionary with same keys as the first
+    but it's items is how many grams do we need to buy.
+    If keys are not the same at both of dictionaries, return null. Otherwise, calculate and return the final price.
+    :param prices: Keys are ingredients names, items are their price for 100 gram
+    :param optionals: List of items (as keys) that we don't want to buy.
+    :param gram_ingredients: Keys as prices dictionary and items as grams we need to buy.
+    :return: none when there is error, else, total price when
+    """
+    if not prices.keys() == gram_ingredients.keys():
```

Comment:
Cant you supply a price that will not be used in this function? seems like you are too harsh in your limitations, without a reason